ff_calculation.py

```diff
@@ -24,10 +24,105 @@
     help="Path to the config file which contains information for the fake factor calculation step.",
 )
 
+parser.add_argument(
+    "--disable-multiprocessing",
+    action="store_true",
+    help="Flag to disable multiprocessing for debugging purposes.",
+)
+
+FF_CALCULATION_FUNCTIONS = {
+    "QCD": FF_QCD.calculation_QCD_FFs,
+    "QCD_subleading": FF_QCD.calculation_QCD_FFs,
+    "Wjets": FF_Wjets.calculation_Wjets_FFs,
+    "ttbar": FF_ttbar.calculation_ttbar_FFs,
+    "ttbar_subleading": FF_ttbar.calculation_ttbar_FFs,
+    "process_fractions": fraction_calculation,
+    "process_fractions_subleading": fraction_calculation,
+}
+
+
+def FF_calculation(
+    config: Dict[str, Union[str, Dict, List]],
+    sample_paths: List[str],
+    output_path: str,
+    process: str,
+    logger: str,
+) -> Dict[str, Union[Dict[str, str], Dict[str, Dict[str, str]]]]:
+    """
+    This function calculates fake factors for a given process or fractions of the processes.
+
+    Args:
+        config: A dictionary with all the relevant information for the fake factor calculation
+        sample_paths: List of file paths where the samples are stored
+        output_path: Path where the generated plots should be stored
+        process: This is relevant for QCD because for the tt channel two different QCD fake factors are calculated, one for each hadronic tau
+        logger: Name of the logger that should be used
+
+    Return:
+        Dictionary where the categories are defined as keys and and the values are the fitted functions (including variations)
+        e.g. corrlib_expressions[CATEGORY_1][CATEGORY_2][VARIATION] if dimension of categories is 2
+    """
+
+    is_fraction = "fraction" in process
+    is_ttbar = "ttbar" in process
+    split_limit = 1 if is_fraction else 2
+
+    process_conf = config[process] if is_fraction else config["target_processes"][process]
+
+    split_variables, split_combinations, split_binnings = ff_func.get_split_combinations(
+        categories=process_conf["split_categories"],
+        binning=process_conf["var_bins"],
+    )
+
+    assert len(split_variables) <= split_limit, "Category splitting is only defined up to 2 dimensions."
```

Comment:
Use and f string with `split_limit` instead of 2, because for fractions it is not 2.

Reply:
I saw now that this is already checked with `ff_func.get_split_combinations`, so I suggest to remove this check here.